shared/libraries/config_protocol/include/config_protocol/config_client_device_impl.h

```diff
@@ -163,6 +165,75 @@ void GenericConfigClientDeviceImpl<TDeviceBase>::handleRemoteCoreObjectInternal(
     Super::handleRemoteCoreObjectInternal(sender, args);
 }
 
+template <class TDeviceBase>
+void GenericConfigClientDeviceImpl<TDeviceBase>::onRemoteUpdate(const SerializedObjectPtr& serialized)
+{
+    ConfigClientComponentBaseImpl<TDeviceBase>::onRemoteUpdate(serialized);
+
+    std::vector<std::string> toRemove;
+    for (const auto& comp : this->components)
+    {
+        const auto id = comp.getLocalId();
+        if (!serialized.hasKey(id))
+        {
+            if (this->defaultComponents.count(id))
+                throw InvalidOperationException{"Serialized remote object does not contain default device component: " + id};
+            
+            toRemove.push_back(id);
+        }
+        else
+        {
+            const auto serObj = serialized.readSerializedObject(id);
+            comp.template asPtr<IConfigClientObject>()->remoteUpdate(serObj);
+        }
+    }
+
+    for (const auto& id : toRemove)
+        this->removeComponentById(id);
+
+    for (const auto& key : serialized.getKeys())
+    {
+        if (this->defaultComponents.count(key))
+            continue;
+
+        if (serialized.getType(key) != ctObject)
```

Comment:
`key` can also be "propValues" or "properties" which I believe it returns ctObject. At least I had issues that I needed to exclude those fields. See commit 15729ff9b5234bbcaf76ead5b6623a62042ef307